pkg/registry/apis/provisioning/jobs/migrate/legacy_resources.go

```diff
@@ -165,11 +171,21 @@ func (r *legacyResourceResourceMigrator) Write(ctx context.Context, key *resourc
 
 	// TODO: this seems to be same logic as the export job
 	// TODO: we should use a kind safe manager here
-	fileName, err := r.resources.CreateResourceFileFromObject(ctx, parsed.Obj, resources.WriteOptions{
+	fileName, err := r.resources.WriteResourceFileFromObject(ctx, parsed.Obj, resources.WriteOptions{
 		Path: "",
 		Ref:  "",
 	})
 
+	// For versioned
+	if r.history != nil {
+		name := parsed.Meta.GetName()
+		previous := r.history[name]
+		if previous != "" {
```

Comment:
can you add a comment on why we do this? why removing the previous file? 

What happens if the previous and the current point to the same thing? We would just write and then remove it, right?